crates/proxy-server/src/proxy_handler.rs

```diff
@@ -13,6 +13,7 @@ pub async fn handle_proxy_client(
     mut stream: TcpStream,
     peer_address: SocketAddr,
     timeout_duration: std::time::Duration,
+    #[cfg_attr(not(target_os = "linux"), allow(unused_variables))]
```

Comment:
I figured it would be better not to have the public interface change depending on platforms, with the downside that `no_splice = true` will be silently ignored on non-Linux platforms.